### PR TITLE
add XDPLua

### PR DESCRIPTION
**Pull request by VictorNogueiraRio**

### CODE REVIEW

**Comment by marcelstanley** [comment]
@VictorNogueiraRio as we discussed, please consider adding 637df06 except for the accidental change to `luarcu`

File: net/core/filter.c [hunk 1]
```diff
@@ -5849,6 +5856,286 @@ static const struct bpf_func_proto bpf_tcp_gen_syncookie_proto = {
...
+	struct xdplua_create_work *lw;
+
+	lw = this_cpu_ptr(&luaworks);
+	if (!lw->init) {
+		lw->init = true;
+		spin_lock(&lw->lock);
+	}
+}
+
+BPF_CALL_2(bpf_lua_dataref, struct xdp_buff *, ctx, int, offset) {
```
Comment (VictorNogueiraRio) [inline net/core/filter.c:5871]: Suggested change, by @lneto 
```
BPF_CALL_2(bpf_lua_dataref, struct xdp_buff *, ctx, int, offset) {
	lua_State *L;
        
        if (ctx->data_end <= offset + ctx->data)
		return -1;

	L = xdplua_lock();
	return ldata_newref(L, ctx->data + offset,
			ctx->data_end - ctx->data - offset);
}
```

File: net/core/filter.c [hunk 2]
```diff
@@ -5849,6 +5856,286 @@ static const struct bpf_func_proto bpf_tcp_gen_syncookie_proto = {
 
 #endif /* CONFIG_INET */
 
+#ifdef CONFIG_XDP_LUA
+
+static inline void verify_and_lock(void) {
```
Comment (VictorNogueiraRio) [inline net/core/filter.c:5861]: Suggested change, by @lneto

``` 
static inline lua_State *xdplua_lock(void) {
       struct xdplua_create_work *lw;

       lw = this_cpu_ptr(&luaworks);
	if (!lw->inuse) {
		lw->inuse = true;
		spin_lock(&lw->lock);
	}

        return lw->L;
}
```

File: include/net/xdp.h [hunk 3]
```diff
@@ -63,13 +63,33 @@ struct xdp_rxq_info {
 	struct xdp_mem_info mem;
 } ____cacheline_aligned; /* perf critical, avoid false-sharing */
 
+#ifdef CONFIG_XDP_LUA
+struct xdplua_create_work {
+	char				lua_script[8192];
+	struct lua_State	*L;
+	struct work_struct	work;
+	spinlock_t		lock;
+	bool			init;
```
Comment (VictorNogueiraRio) [inline include/net/xdp.h:72]: s/init/inuse/

**Comment by VictorNogueiraRio** [comment]
Obsoleted

**Pull request closed without merging**